### PR TITLE
Default to libvirt firmware being UEFI

### PR DESCRIPTION
**Pull request by stbenjam**
This changes the default firmware to always be UEFI.

### CODE REVIEW

**Comment by stbenjam** [comment]
https://github.com/metal3-io/baremetal-operator/pull/602 changes the BMO default to UEFI. I don't think the two PR's need to be tested together... PXE should work for UEFI boot as currently boot_capabilities should be empty.

File: lib/network.sh [hunk 1]
```diff
@@ -38,12 +38,12 @@ export POD_CIDR=${POD_CIDR:-"192.168.0.0/18"}
 PROVISIONING_IPV6=${PROVISIONING_IPV6:-false}
 IPV6_ADDR_PREFIX=${IPV6_ADDR_PREFIX:-"fd2e:6f44:5dd8:b856"}
 
+export LIBVIRT_FIRMWARE=${LIBVIRT_FIRMWARE:uefi}
```
Comment (maelk) [inline lib/network.sh:41]: there is a typo here, you're missing a "-" to default it should be ":-"

Comment (stbenjam) [reply lib/network.sh]: Thanks!

**Comment by hardys** [comment]
Tested locally and this works fine for me - I see the following error in CI:

```
2020-07-23 12:55:58.011+0000: Domain id=12 is tainted: host-cpu
libvirt:  error : libvirtd quit during handshake: Input/output error
2020-07-23 12:55:58.023+0000: shutting down, reason=failed
```

So I wonder if this is a test-infra issue?

/test-integration

**Comment by stbenjam** [comment]
> Tested locally and this works fine for me - I see the following error in CI:
> 2020-07-23 12:55:58.011+0000: Domain id=12 is tainted: host-cpu
libvirt:  error : libvirtd quit during handshake: Input/output error
2020-07-23 12:55:58.023+0000: shutting down, reason=failed
>
> So I wonder if this is a test-infra issue?

Yea, I don't know why but when I was looking in the logs, I saw a lot of network errors in CI. I couldn't figure out what was going on, I was hoping https://github.com/metal3-io/project-infra/pull/129 would help a bit.

**Comment by maelk** [comment]
This is really weird error. We have not seen this error in the master runs of the CI. Maybe we're hitting something with libvirt and nested virtualization ? we have never run with uefi in CI.

**Comment by maelk** [comment]
/test-integration
Investigating the CI run

**Comment by stbenjam** [comment]
> This is really weird error. We have not seen this error in the master runs of the CI. Maybe we're hitting something with libvirt ? we have never run with uefi in CI.

It could be, do you know what version of edk2 is on these boxes? Not sure the name of the package on ubuntu, but its edk2-ovmf on Fedora. We've been using it in OpenShift for a while on the control plane and normally PXE-based provisioning just works the same way.

**Comment by maelk** [comment]
I can see this error when trying to start it manually 
```
error: Failed to start domain node_0
error: Path '/usr/share/OVMF/OVMF_CODE.secboot.fd' is not accessible: No such file or directory
```

**Comment by maelk** [comment]
turns out that there is a file /usr/share/ovmf/OVMF.fd , and when modifying the vm to use that one, it starts properly.

**Comment by maelk** [comment]
Any chance that https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/libvirt/templates/baremetalvm.xml.j2#L11 and following line are incorrect for other distributions than Fedora ?

**Comment by maelk** [comment]
apparently if we were running Ubuntu 20.04, that file would be there. It may be time to upgrade! I didn't check CentOS though

**Comment by maelk** [comment]
https://github.com/metal3-io/project-infra/issues/130 to trackthe Ubuntu upgrade.

**Comment by fmuyassarov** [comment]
> > Tested locally and this works fine for me - I see the following error in CI:
> > 2020-07-23 12:55:58.011+0000: Domain id=12 is tainted: host-cpu
> > libvirt:  error : libvirtd quit during handshake: Input/output error
> > 2020-07-23 12:55:58.023+0000: shutting down, reason=failed
> > So I wonder if this is a test-infra issue?
> 
> Yea, I don't know why but when I was looking in the logs, I saw a lot of network errors in CI. I couldn't figure out what was going on, I was hoping [metal3-io/project-infra#129](https://github.com/metal3-io/project-infra/pull/129) would help a bit.

https://github.com/metal3-io/project-infra/pull/129 is merged now, in case if you want to check yamls.

**Comment by stbenjam** [comment]
The EDK2 firmware is such a buggy mess in my experience. The firmwares also are named very inconsistently across distros. It makes total sense the paths are different everywhere, we'll have to account for that, but even if we do sounds like the one in Ubuntu is busted?

**Comment by maelk** [comment]
well I did not get the machine to boot with the existing one in 18.04. I'm trying to setup the upgrade for 20.04

**Comment by stbenjam** [comment]
Not clear to me why CentOS CI is failing though -- edk2-ovmf-20190829git37eef91017ad-9.el8.noarch is installed there, and it has /usr/share/OVMF/OVMF_CODE.secboot.fdf

**Comment by maelk** [comment]
I can investigate that. it might be a different error

**Comment by maelk** [comment]
unrelated error in centos tests
/test-centos-integration

**Comment by maelk** [comment]
Here is the Ubuntu upgrade PR : https://github.com/metal3-io/metal3-dev-env/pull/394

**Comment by maelk** [comment]
@stbenjam I looked into the centos job. The problem is different there indeed. The IPMI node succeeds, but the redfish+virtualmedia fails. I don't have much more details yet.

**Comment by maelk** [comment]
Ubuntu tests should now be running with Ubuntu 20.04, that has the proper OVMF files. let's see if that solves the issue

**Comment by maelk** [comment]
@stbenjam you would need to modify the 03_* script to generate BMHs with uefi set.

**Pull request closed without merging**